### PR TITLE
Docs - Add draft whitelist sso doc

### PR DESCRIPTION
**Pull request by gongjason**
Continued the work on https://github.com/firezone/firezone/pull/995. Moved to a new PR due to too many merge conflicts.

### CODE REVIEW

File: docs/docs/user-guides/use-cases/nat-gateway/README.mdx [hunk 1]
```diff
@@ -31,6 +32,9 @@ through a Firezone server on an EC2 instance. In this case Firezone is acting as
 a network proxy or NAT gateway to provide a single public egress IP for all the
 devices connected to it.
 
+In this example the protected resource and Firezone are
+in separate VPC regions.
```
Comment (jamilbk) [inline docs/docs/user-guides/use-cases/nat-gateway/README.mdx]:
```suggestion
In this example the protected resource and Firezone are in separate VPC regions.
```

File: docs/docs/user-guides/use-cases/nat-gateway/README.mdx [hunk 2]
```diff
@@ -15,12 +18,10 @@ This is commonly used in the following scenarios:
...
 self-hosted web app to a single whitelisted static IP running Firezone.
-In this example the protected resource and Firezone are
-in separate VPC regions.
 
-This arrangement is commonly done in place of maintaining an IP whitelist for
-multiple end users, which may become labor intensive to manage as the access
-list grows.
+This is commonly done in place of maintaining an IP whitelist for
+multiple end users, which may become labor-intensive to manage as the
+access list grows.
```
Comment (jamilbk) [inline docs/docs/user-guides/use-cases/nat-gateway/README.mdx]:
```suggestion
This is commonly done in place of maintaining an IP whitelist for
multiple team members, which becomes impossible to manage as the
access list grows and team members' IP addresses change.
```

File: docs/docs/user-guides/use-cases/nat-gateway/README.mdx [hunk 3]
```diff
@@ -60,13 +64,9 @@ party to allow traffic from the static IP set in Step 1 (in this case `52.202.88
 
 ### Step 3 - Route traffic to the protected resource through the VPN server
 
-By default all traffic from users will be routed through the VPN server,
+By default, all traffic from users will be routed through the VPN server
```
Comment (jamilbk) [inline docs/docs/user-guides/use-cases/nat-gateway/README.mdx]:
```suggestion
By default, all traffic from team members will be routed through Firezone
```

File: docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx [hunk 4]
```diff
@@ -0,0 +1,88 @@
+---
+title: Restrict SSO to VPN Users
+sidebar_position: 1
+description: Configure your identity provider to whitelist traffic from Firezone's WireGuard® based secure access platform.
+---
+
+# Restrict application authentication to VPN users
+
+Some identity providers support restricting SSO functionality
+to a pre-defined network zone (IP range). You can set this to the IP of
```
Comment (jamilbk) [inline docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx]: typo

```suggestion
Some identity providers support restricting SSO access to a predefined CIDR range or list of IP addresses. You can set this to the IP address of
```

File: docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+your Firezone gateway, so authentication is restricted to users
+connected to the VPN.
```
Comment (jamilbk) [inline docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx]:
```suggestion
your Firezone gateway to restrict SSO to users connected to Firezone.
```

File: docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx [hunk 6]
```diff
@@ -0,0 +1,88 @@
+---
+title: Restrict SSO to VPN Users
```
Comment (jamilbk) [inline docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx]:
```suggestion
title: Restrict SSO to Firezone Users
```

File: docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+:::
+
+:::caution
+Sign-on policies are assigned on an application level. We advise
+excluding the Firezone application from the IP restriction policy above if Okta
+is being used for Firezone authentication.
+
+With
+[periodic authentication](../../authenticate#enforce-periodic-re-authentication)
+enabled, you may be locked out Okta entirely when your VPN session
```
Comment (jamilbk) [inline docs/docs/user-guides/use-cases/nat-gateway/_whitelist-sso.mdx]:
```suggestion
enabled, you may be locked out of Okta entirely when your VPN session
```

**Review by jamilbk — CHANGES_REQUESTED** [review]

**Comment by gongjason** [comment]
Oops, was renaming the branch to comply with conventional commits, but just realized that closes the PR. Will just have proper naming of branches going fwd to not lose the discussion here for this PR and https://github.com/firezone/firezone/pull/1187.

**Review by jamilbk — APPROVED** [review]

**Pull request merged**